### ISSUE TITLE
What is recommended system configuration for Deep Learning?

### ISSUE THREAD

**Issue by spurani** [tensorflow/tensorflow#38690]
Basically I am having Acer Nitro 50 Desktop with system configuration
Processor: Intel® Core™ i5-8400 CPU @ 2.80GHz × 6
Graphics: GeForce GTX 1050/PCIe/SSE2 (2 GB)
Memory(RAM): 8GB DDR4 Memory

I am working with **tensorflow_gpu-1.12.0 | python 3.5 | GCC 4.8 | Bazel 0.15.0 | cudnn 7 | Cuda 9.0** to train a faster_rcnn_inception_v2 model on my custom dataset with 10.4 GB(65000 images) of training data and 533.4 MB(3333) of validation data for object detection for 600k epochs(num_steps it will be training at 640 x 640 resolution). I am training and validation model on 8 classes. So when I am training the model the accuracy(map) is not increasing and loss is not decreasing after a while. After the completion of the training successfully I ran the model on various images and noticed couple of things.

1)    Less accuracy
 2)   In some images objects does not get detected at all no bounding boxes. while in some multiclass object detection is spot on perfectly all objects gets detected but with less accuracy around 60 - 75.
 3)   From second point above If I train a separate model with less images and less number of classes(3 or 4) it works well but with decent amount of accuracy around 75 - 95

Question 1) What would be ideal system to work comfortably with deep learning? and get desired amount of results

So based on my system configuration should I keep on training or invest into new system which will be lot faster and accurate in deep learning. It would be really helpful for me to continue my learning if anyone could suggest or recommend something

**Comment by ravikyram** [comment]
@spurani 

Please, see hardware and software requirements from [here](https://www.tensorflow.org/install/gpu).Also, please see tested build configurations for [linux](https://www.tensorflow.org/install/source#linux), [mac](https://www.tensorflow.org/install/source#gpu_2), [windows](https://www.tensorflow.org/install/source_windows#gpu).Thanks!

**Comment by ravikyram** [comment]
@spurani 

Any update on this issue please. Thanks!

**Comment by spurani** [comment]
Sorry for my late reply I have installed tensorflow using https://www.tensorflow.org/install/gpu but the hardware requirements are kind of vague it would be helpful If I would get to know that a specific hardware is recommended to run the tensorflow models.  thanks and cheers

**Comment by ravikyram** [comment]
@spurani 

Please, go through [hardware requirements](https://www.tensorflow.org/install/pip?lang=python3#hardware-requirements) and see if it helps you.Thanks!

**Comment by spurani** [comment]
![image](https://user-images.githubusercontent.com/28580888/80699931-de7c8f80-8aaa-11ea-96b3-72775b0e1636.png)
![image](https://user-images.githubusercontent.com/28580888/80700150-33b8a100-8aab-11ea-9463-00f61632015a.png)

So this are the images of learning rate and total loss which clearly indicates that my model is not training efficiently. It would be really helpful if I could get some advice on how should I make it work?

**Comment by gowthamkpr** [comment]
@spurani There might be many factors that might be impacting your accuracy. As answered in this **[question](https://stackoverflow.com/questions/35293468/does-reducing-classes-in-a-classification-method-improve-accuracy), having less number of classes will increase your accuracy and having more number of classes will most likely decrease your accuracy. This doesn't mean you have to change the system configuration to get good results. 
Also you can refer to the following question [here](https://stats.stackexchange.com/questions/348584/do-more-object-classes-increase-or-decrease-the-accuracy-of-object-detection) to understand more about your problem 

Also I would recommend you to post your issue [here](https://github.com/tensorflow/models/issues) and also in stackoverflow where there is a wider community to respond. Thanks!

**Comment by spurani** [comment]
Thank you @gowthamkpr  for those references those might come handy but they are pretty hard to understand if you come across any tutorial where is model is trained on multiple categories and huge amount of dataset can you please forward it? As the images in MS COCO dataset does not have same size do I need to resize before training change the image_resizer function in faster_rcnn_inception_v2_coco.config file(above in previous question)? How they would have trained a model on 90 different categories with huge dataset? It would be great if anyone can guide me through this it will reaally help me continue my learning on machine learning thank you

**Comment by gowthamkpr** [comment]
@spurani I will do that and also As mentioned in the above comment, please post this issue in tensorflow/models or  stackoverflow as there is a wider community to respond. Thanks!
